KeyError on master in TestExcelWrtiter

https://travis-ci.org/pandas-dev/pandas/jobs/424485100 isn't master, but it's seeminglyon other PRs. e.g. https://travis-ci.org/pandas-dev/pandas/jobs/424480841#L2253

```
______________ TestExcelWriter.test_tsframe[openpyxl-.xlsx-True] _______________
[gw0] darwin -- Python 3.5.6 /Users/travis/miniconda3/envs/pandas/bin/python
self = <pandas.tests.io.test_excel.TestExcelWriter object at 0x1148ad9b0>
merge_cells = True, engine = 'openpyxl', ext = '.xlsx'
    def test_tsframe(self, merge_cells, engine, ext):
        df = tm.makeTimeDataFrame()[:5]
    
>       df.to_excel(self.path, 'test1')
pandas/tests/io/test_excel.py:1153: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pandas/core/frame.py:1749: in to_excel
    engine=engine)
pandas/io/formats/excel.py:658: in write
    freeze_panes=freeze_panes)
pandas/io/excel.py:1419: in write_cells
    xcell.value, fmt = self._value_with_fmt(cell.val)
../../../miniconda3/envs/pandas/lib/python3.5/site-packages/openpyxl/cell/cell.py:291: in value
    self._bind_value(value)
../../../miniconda3/envs/pandas/lib/python3.5/site-packages/openpyxl/cell/cell.py:193: in _bind_value
    self._set_time_format(value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <Cell 'test1'.A2>, value = Timestamp('2000-01-03 00:00:00', freq='B')
    def _set_time_format(self, value):
        """Set number format for Python date or time"""
        fmts = {
            datetime.datetime:numbers.FORMAT_DATE_DATETIME,
            datetime.date:numbers.FORMAT_DATE_YYYYMMDD2,
            datetime.time:numbers.FORMAT_DATE_TIME6,
            datetime.timedelta:numbers.FORMAT_DATE_TIMEDELTA,
                }
>       self.number_format = fmts[type(value)]
E       KeyError: <class 'pandas._libs.tslibs.timestamps.Timestamp'>
```

Trying to reproduce locally now.

--- a/ci/appveyor-27.yaml
+++ b/ci/appveyor-27.yaml
@@ -13,7 +13,7 @@ dependencies:
   - matplotlib
   - numexpr
   - numpy=1.12*
-  - openpyxl
+  - openpyxl=2.5.5
   - pytables
   - python=2.7.*
   - pytz

--- a/ci/appveyor-36.yaml
+++ b/ci/appveyor-36.yaml
@@ -10,7 +10,7 @@ dependencies:
   - matplotlib
   - numexpr
   - numpy=1.14*
-  - openpyxl
+  - openpyxl=2.5.5
   - pyarrow
   - pytables
   - python-dateutil

--- a/ci/circle-27-compat.yaml
+++ b/ci/circle-27-compat.yaml
@@ -8,7 +8,7 @@ dependencies:
   - jinja2=2.8
   - numexpr=2.4.4 # we test that we correctly don't use an unsupported numexpr
   - numpy=1.9.3
-  - openpyxl
+  - openpyxl=2.5.5
   - psycopg2
   - pytables=3.2.2
   - python-dateutil=2.5.0

--- a/ci/circle-36-locale.yaml
+++ b/ci/circle-36-locale.yaml
@@ -13,7 +13,7 @@ dependencies:
   - nomkl
   - numexpr
   - numpy
-  - openpyxl
+  - openpyxl=2.5.5
   - psycopg2
   - pymysql
   - pytables

--- a/ci/circle-36-locale_slow.yaml
+++ b/ci/circle-36-locale_slow.yaml
@@ -14,7 +14,7 @@ dependencies:
   - nomkl
   - numexpr
   - numpy
-  - openpyxl
+  - openpyxl=2.5.5
   - psycopg2
   - pymysql
   - pytables

--- a/ci/requirements-optional-conda.txt
+++ b/ci/requirements-optional-conda.txt
@@ -12,7 +12,7 @@ lxml
 matplotlib
 nbsphinx
 numexpr
-openpyxl
+openpyxl=2.5.5
 pyarrow
 pymysql
 pytables

--- a/ci/requirements-optional-pip.txt
+++ b/ci/requirements-optional-pip.txt
@@ -14,7 +14,7 @@ lxml
 matplotlib
 nbsphinx
 numexpr
-openpyxl
+openpyxl=2.5.5
 pyarrow
 pymysql
 tables

--- a/ci/travis-35-osx.yaml
+++ b/ci/travis-35-osx.yaml
@@ -12,7 +12,7 @@ dependencies:
   - nomkl
   - numexpr
   - numpy=1.10.4
-  - openpyxl
+  - openpyxl=2.5.5
   - pytables
   - python=3.5*
   - pytz

--- a/ci/travis-36-doc.yaml
+++ b/ci/travis-36-doc.yaml
@@ -22,7 +22,7 @@ dependencies:
   - notebook
   - numexpr
   - numpy=1.13*
-  - openpyxl
+  - openpyxl=2.5.5
   - pandoc
   - pyqt
   - pytables

--- a/ci/travis-36-slow.yaml
+++ b/ci/travis-36-slow.yaml
@@ -10,7 +10,7 @@ dependencies:
   - matplotlib
   - numexpr
   - numpy
-  - openpyxl
+  - openpyxl=2.5.5
   - patsy
   - psycopg2
   - pymysql

--- a/ci/travis-36.yaml
+++ b/ci/travis-36.yaml
@@ -18,7 +18,7 @@ dependencies:
   - nomkl
   - numexpr
   - numpy
-  - openpyxl
+  - openpyxl=2.5.5
   - psycopg2
   - pyarrow
   - pymysql

